tools/c7n_azure/c7n_azure/function_package.py

```diff
@@ -31,6 +28,25 @@
 from c7n_azure.dependency_manager import DependencyManager
 from c7n_azure.session import Session
 
+from c7n.mu import PythonPackageArchive
+from c7n.utils import local_session
+
+
+class AzurePythonPackageArchive(PythonPackageArchive):
+    def __init__(self, modules=(), cache_file=None):
+        super(AzurePythonPackageArchive, self).__init__(modules, cache_file)
+        self.package_time = time.gmtime()
+
+    def create_zinfo(self, fname):
+        """
+        In Dedicated App Service Plans - Functions are updated via KuduSync
+        The KuduSync uses the modified time and file size to determine if a file has changed
```

Comment:
if file size is being considered, wouldn't the file size change should work alone? or is it only looked at after timestamp?

Reply:
They check both but it is very common for size to remain the same for an update (bugs people were hitting were while updating the minutes value of a cron spec with no size change)

https://github.com/projectkudu/KuduSync/blob/d0b9dec32fc87d49d261807e13bd99d57dc9bbf2/lib/FileInfo.ts#L29

Reply:
ic, i guess this is partly that some parts of the function configuration is actually are actually part of the function deployment package interpreted by the host environment, and the hash on that config changes (cron expression change) but not the size.

the flip side here is that one would always end up deploying even on no-op changes afaics with this change because effectively forces newer timestamps on every run. for gcp/aws we've taken pains to avoid that (ie no change == no write api calls), but if thats the only path here, seems fine, and the change here is isolated to azure function path.

Reply:
I opened an issue to transition to WEBSITE_RUN_FROM_PACKAGE #4430